torchx/components/component_test_base.py

```diff
@@ -58,3 +78,43 @@ def validate(self, module: ModuleType, function_name: str) -> None:
         # this will raise an exception and the test will fail
         with self.assertRaises(SystemExit):
             _ = _create_args_parser(component_def.fn).parse_args(["--help"])
+
+    def run_component(
+        self,
+        component: Callable[..., AppDef],
+        args: Optional[Dict[str, Any]] = None,
```

Comment:
wonder if it's slightly cleaner to use `**args` but that also mixes components vs run_component args so either is fine